lib/util/clone.ts

```diff
@@ -1,11 +1,11 @@
-import { quickStringify } from './stringify';
-
 /**
  * Creates a deep clone of an object.
- * @deprecated Use {@link structuredClone} instead.
  * @param input The object to clone.
  */
-export function clone<T>(input: T | null = null): T {
-  const stringifiedInput = quickStringify(input);
-  return stringifiedInput ? JSON.parse(stringifiedInput) : null;
+export function clone<T = unknown>(input: T): T {
+  const res = structuredClone(input);
+  if (typeof input === 'object' && input !== null) {
+    Object.setPrototypeOf(res, Object.getPrototypeOf(input));
+  }
```

Comment:
Are there any implications using `Object.setPrototypeOf` that we should be aware of? It seems like it is making some properties that were previously optional, or not part of the `ReleaseResult` type `undefined`.

MDN has a pretty strong warning about `Object.setPrototypeOf`[^1]

>Warning: Changing the [[Prototype]] of an object is, by the nature of how modern >JavaScript engines optimize property accesses, currently a very slow operation in every >browser and JavaScript engine. In addition, the effects of altering inheritance are >subtle and far-flung, and are not limited to the time spent in the >Object.setPrototypeOf(...) statement, but may extend to any code that has access to any >object whose [[Prototype]] has been altered. You can read more in [JavaScript engine >fundamentals: optimizing prototypes](https://mathiasbynens.be/notes/prototypes).
>
>Because this feature is a part of the language, it is still the burden on engine >developers to implement that feature performantly (ideally). Until engine developers >address this issue, if you are concerned about performance, you should avoid setting >the [[Prototype]] of an object. Instead, create a new object with the desired >[[Prototype]] using [Object.create()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/create).

[^1]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/setPrototypeOf

Reply:
Here I'm resetting whatever prototype `structuredClone` sets internally. Without this, `is.plainObject()` stops working on cloned objects.

Reply:
set prototype is a bad idea, makes code too slow again. we probably should stick to JSON serialization again instead of structured clone.

Reply:
In this PR?

Reply:
First step wrap, second step change implementation?